src/command/info/info.c

```diff
@@ -350,6 +358,271 @@ backupList(VariantList *backupSection, InfoBackup *info, const String *backupLab
     FUNCTION_TEST_RETURN_VOID();
 }
 
+/***********************************************************************************************************************************
+Generate the wal archives chain in the given interval and parse .history file if needed.
+***********************************************************************************************************************************/
+static void
+generateNeededArchivesList(
+    StringList *neededArchivesList, const unsigned int pgVersion, const String *stanza, const String *archiveId,
+    const String *archiveStart, const String *archiveStop)
+{
+    FUNCTION_TEST_BEGIN();
+        FUNCTION_TEST_PARAM(STRING_LIST, neededArchivesList);
+        FUNCTION_TEST_PARAM(UINT, pgVersion);
+        FUNCTION_TEST_PARAM(STRING, archiveStart);
+        FUNCTION_TEST_PARAM(STRING, archiveStop);
+    FUNCTION_TEST_END();
+
+    ASSERT(neededArchivesList != NULL);
+    ASSERT(archiveStart != NULL);
+    ASSERT(archiveStop != NULL);
+
+    // Until we find a way to get the WAL segment size from e.g. the archive.info, let's use the PG default size
+    unsigned int PG_WAL_SEGMENT_SIZE_DEFAULT = 16 * 1024 * 1024;
+
+    // Get .history file if timelines don't match
+    uint32_t startTimeline = (uint32_t)strtol(strZ(strSubN(archiveStart, 0, 8)), NULL, 16);
+    uint32_t stopTimeline = (uint32_t)strtol(strZ(strSubN(archiveStop, 0, 8)), NULL, 16);
+
+    // Store history content: WAL segments and target timeline
+    StringList *timelineSwitchWal = strLstNew();
+    KeyValue *targetTimelines = kvNew();
+
+    if(stopTimeline > startTimeline)
```

Comment:
If I'm understanding correctly, the problem that is trying to be solved is determining if all the WAL exists in the repo to get from a backup through the latest timeline. If that is the case, passing the archive wal start/stop for a specific backup to this function will never satisfy the stopTimeline > startTimeline condition since a single backup will never cross timelines - a set will, but not a backup within a set. 

As an example, if you have docker and can run the pgbackrest docs though the pitr section:
`/<YOUR DIR>/doc/doc.pl --out=html --include=user-guide --require=/pitr --var=encrypt=n --no-cache`
then 
`docker exec -it doc-pg-primary bash`
you can see that the `info` command might produce:

```
   db (current)
        wal archive min/max (12-1): 000000010000000000000001/000000030000000000000007

        full backup: 20200810-171426F
            timestamp start/stop: 2020-08-10 17:14:26 / 2020-08-10 17:14:40
            wal start/stop: 000000010000000000000002 / 000000010000000000000002
            database size: 23.4MB, backup size: 23.4MB
            repository size: 2.8MB, repository backup size: 2.8MB

        diff backup: 20200810-171426F_20200810-171442D
            timestamp start/stop: 2020-08-10 17:14:42 / 2020-08-10 17:14:44
            wal start/stop: 000000010000000000000003 / 000000010000000000000003
            database size: 23.4MB, backup size: 8.3KB
            repository size: 2.8MB, repository backup size: 426B
            backup reference list: 20200810-171426F

        diff backup: 20200810-171426F_20200810-171445D
            timestamp start/stop: 2020-08-10 17:14:45 / 2020-08-10 17:14:47
            wal start/stop: 000000010000000000000004 / 000000010000000000000004
            database size: 23.4MB, backup size: 8.3KB
            repository size: 2.8MB, repository backup size: 425B
            backup reference list: 20200810-171426F

        incr backup: 20200810-171426F_20200810-171459I
            timestamp start/stop: 2020-08-10 17:14:59 / 2020-08-10 17:15:03
            wal start/stop: 000000020000000000000006 / 000000020000000000000006
            database size: 23.4MB, backup size: 2.2MB
            repository size: 2.8MB, repository backup size: 232.5KB
            backup reference list: 20200810-171426F, 20200810-171426F_20200810-171445D
```

(you can use --out=json to get the kv structured output). So a single backup archive `wal start/stop` will never cross a timeline. But 20200810-171426F_20200810-171459I relies on prior backups (backup reference list) so the set does cross timelines - but that is not the information that is being passed to the `generateNeededArchivesList()` function.

It's a little hard to see what is the expected outcome of the patch without tests so my apologies in advance if I'm misunderstanding your intent.

Reply:
Well, indeed, for the backups it's more unlikely that there will be a timeline switch. 
But this function is also used (a bit further, line 586) to check archives consistency for (in your example) `wal archive min/max (12-1): 000000010000000000000001/000000030000000000000007` where a timeline switch occurred :-)

A good test is to retention-archive-type to diff, so archives between the 1st full and 1st diff will be expired. That's why we need to check globally between min (and change the min according the archive retention policy) and max wal but also between each backup start/stop location.

I'll look to add some tests if it's easier to review then ;-)

Reply:
Yes, I think providing tests would be helpful. Also, take care on your historyExp as it will not only match the `2	0/7000000` in 
`2	0/7000000	before 2000-01-01 00:00:00+00`
but also 
```
2 01111/7000000
0111/012345766985640856045968409
```
etc

Reply:
I don't think we need full test coverage here, yet, but a few tests would definitely help us review the code.